test_requirements.txt

```diff
@@ -1,6 +1,9 @@
 flake8==7.0.0
 flake8-docstrings==1.7.0
+numpy==1.24.4; python_version<'3.10'
+pandas==2.0.3; python_version<'3.10'
+numpy>=2.0.2,<=2.1.0; python_version>='3.10'
+pandas==2.3.0; python_version>='3.10'
```

Comment:
There are some unfortunate import errors if we're not careful about numpy/pandas/Pint combinations